File: java/src/test/java/com/alxarafe/domain/catalogue/entity/ItemFamilyTest.java
package com.alxarafe.domain.catalogue.entity;

import com.alxarafe.app.domain.catalogue.entity.ItemFamily;
import com.alxarafe.app.domain.catalogue.valueobject.ItemFamilyCode;
import com.alxarafe.app.domain.catalogue.valueobject.ItemFamilyId;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class ItemFamilyTest {

    @Test
    void create() {
        var id = new ItemFamilyId("018e4e3a-3e7b-7b3e-8000-000000000001");
        var code = new ItemFamilyCode("ELECTRONICS");
        var family = new ItemFamily(id, code, "Consumer Electronics");

        assertEquals(id, family.id());
        assertEquals(code, family.code());
        assertEquals("Consumer Electronics", family.name());
    }

    @Test
    void nullName() {
        assertThrows(NullPointerException.class, () -> new ItemFamily(
                new ItemFamilyId("018e4e3a-3e7b-7b3e-8000-000000000001"),
                new ItemFamilyCode("ELEC"),
                null));
    }

    @Test
    void blankName() {
        assertThrows(IllegalArgumentException.class, () -> new ItemFamily(
                new ItemFamilyId("018e4e3a-3e7b-7b3e-8000-000000000001"),
                new ItemFamilyCode("ELEC"),
                ""));
    }
}

File: php/tests/Unit/Domain/Catalogue/Entity/ItemFamilyTest.php
<?php

declare(strict_types=1);

namespace Tests\Unit\Domain\Catalogue\Entity;

use Alxarafe\App\Domain\Catalogue\Entity\ItemFamily;
use Alxarafe\App\Domain\Catalogue\ValueObject\ItemFamilyCode;
use Alxarafe\App\Domain\Catalogue\ValueObject\ItemFamilyId;
use Alxarafe\App\Domain\Catalogue\ValueObject\StorageAttributeCode;
use InvalidArgumentException;
use PHPUnit\Framework\TestCase;

final class ItemFamilyTest extends TestCase
{
    public function testCreate(): void
    {
        $id = new ItemFamilyId('018e4e3a-3e7b-7b3e-8000-000000000001');
        $code = new ItemFamilyCode('ELECTRONICS');
        $family = new ItemFamily($id, $code, 'Consumer Electronics');

        self::assertTrue($id->equals($family->id()));
        self::assertTrue($code->equals($family->code()));
        self::assertSame('Consumer Electronics', $family->name());
    }

    public function testEmptyName(): void
    {
        $this->expectException(InvalidArgumentException::class);
        new ItemFamily(
            new ItemFamilyId('018e4e3a-3e7b-7b3e-8000-000000000001'),
            new ItemFamilyCode('ELEC'),
            '',
        );
    }

    public function testDuplicateAttributeCodes(): void
    {
        $this->expectException(InvalidArgumentException::class);
        new ItemFamily(ItemFamilyId::generate(), new ItemFamilyCode('FOOD'), 'Food', [
            new StorageAttributeCode('FOOD'),
            new StorageAttributeCode('FOOD'),
        ]);
    }
}
